user-domain-service/app/decorators.py

```diff
@@ -0,0 +1,16 @@
+from flask import request, Response
+
+def authenticate_user (func):
+    def wrapper_authenticate(*args, **kwargs):
+
+        username = request.form['username']
```

Comment:
Since this service would likely be called via ajax or something similar rather than server side rendering it would probably be best practice to pass these values as application/json data rather than as form data.